packages/core/src/pages/s.tsx

```diff
@@ -48,21 +55,21 @@ type UseSearchParams = {
 }
 
 const useSearchParams = ({ sort: defaultSort }: UseSearchParams) => {
-  const [params, setParams] = useState<SearchState | null>(null)
   const { asPath } = useRouter()
 
-  useEffect(() => {
+  const hrefState = useMemo(() => {
     const url = new URL(asPath, 'http://localhost')
 
     const shouldUpdateDefaultSort = defaultSort && !url.searchParams.has('sort')
     if (shouldUpdateDefaultSort) {
       url.searchParams.set('sort', defaultSort)
     }
 
-    setParams(parseSearchState(url))
+    const newState = parseSearchState(url)
+    return formatSearchState(newState).href
   }, [asPath, defaultSort])
 
-  return params
+  return useMemo(() => parseSearchState(new URL(hrefState)), [hrefState])
```

Comment:
`hrefState` is useless by itself, right? I'd move this `useMemo` computations inside the previous one and return it directly.

Reply:
done [here](https://github.com/vtex/faststore/pull/1967/commits/307ca6b94a2f554a460e4f510760dadcc82b08e9)